### PR TITLE
NavDrawer: Include survey info

### PR DESCRIPTION
**Pull request by scolsen**
Updates the layout of the navigation drawer to include information about the active survey and gets us closer to the latest mocks.

Towards #2136 

Screenshot 2024-03-15 at 12 59 00 PM
(the black rectangle isn't in the UI, I just redacted my email)

@sufyanAbbasi

### CODE REVIEW

**Review by gino-m — CHANGES_REQUESTED** [review]
Thanks @scolsen ! Would you be able to approximate the spacing and styling in [the Figma design](https://www.figma.com/file/r2uGtyOiWoDa29za7DNuUc/Ground-Acacia-UX?type=design&node-id=5153-17050&mode=dev) more accurately?  Since this is such a prominent screen the extra polish will have a bigger impact. 

Also note some test appear to be failing, but can't tell which from GCB.

File: ground/src/main/res/layout/nav_drawer_header.xml [hunk 1]
```diff
@@ -50,6 +54,57 @@
       style="@style/NavDrawerHeader.SecondaryText"
       android:text="@{user.email}"
       tools:text="User email" />
-
+    <com.google.android.material.divider.MaterialDivider
```
Comment (sufyanAbbasi) [inline ground/src/main/res/layout/nav_drawer_header.xml:57]: Move the divider below "switch survey" button.

**Comment by sufyanAbbasi** [comment]
Everything looks good to me, but following up on styles, I have some ideas where we can get the most bang for our buck.

I think the failing tests are most likely due to the change in nav items in `HomeScreenFragmentTest`, since I found those to fail when I made a change to the nav recently.

File: ground/src/main/res/layout/nav_drawer_header.xml [hunk 2]
```diff
@@ -50,6 +54,57 @@
...
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content" />
+    <LinearLayout
+      style="@style/NavDrawerHeader"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="16dp"
+      android:fitsSystemWindows="true"
+      android:orientation="vertical">
```
Comment (sufyanAbbasi) [inline ground/src/main/res/layout/nav_drawer_header.xml:66]: Throw in 16dp padding start and end to make the nav content flush.

File: ground/src/main/res/layout/nav_drawer_header.xml [hunk 3]
```diff
@@ -50,6 +54,57 @@
...
+         android:layout_width="match_parent"
+         android:layout_height="wrap_content"
+         android:orientation="vertical">
+      <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:text="@{survey.title}"
+        android:textStyle="bold"
+        tools:text="Deforestation areas" />
```
Comment (sufyanAbbasi) [inline ground/src/main/res/layout/nav_drawer_header.xml:91]: This seems out of place, as well as the tools text below: "A survey about farms."

File: ground/src/main/res/layout/nav_drawer_header.xml [hunk 4]
```diff
@@ -50,6 +54,57 @@
...
       tools:text="User email" />
-
+    <com.google.android.material.divider.MaterialDivider
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content" />
+    <LinearLayout
+      style="@style/NavDrawerHeader"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="16dp"
```
Comment (sufyanAbbasi) [inline ground/src/main/res/layout/nav_drawer_header.xml:64]: Also throw in a marginBottom `16dp`

**Review by sufyanAbbasi — CHANGES_REQUESTED** [review]

File: ground/src/main/res/layout/nav_drawer_header.xml (hunk 3 above)
Comment (scolsen) [reply ground/src/main/res/layout/nav_drawer_header.xml:91]: Changed it to more generic text, I was just using these to check out a semi-realistic preview of the layout in android studio

File: ground/src/main/res/layout/nav_drawer_header.xml [hunk 5]
```diff
@@ -50,6 +54,57 @@
...
+        android:layout_height="wrap_content"
+        android:text="@string/current_survey"
+        app:drawableStartCompat="@drawable/ic_content_paste"
+        tools:text="Current survey" />
+      <TextView
+        android:id="@+id/no_surveys_info"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/no_survey_selected"
+        android:visibility="gone"
```
[the remark this answers is not in the document]
Comment (gino-m) [reply ground/src/main/res/layout/nav_drawer_header.xml:78]: Adding it now.

**Comment by gino-m** [comment]
@scolsen  I couldn't write to so I moved your and my commits into #2382. Hope it's ok!

**Pull request closed without merging**